src/app/admin/admin-node-fees/admin-node-add-fee/admin-node-add-fees.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component, EventEmitter, Input, OnInit, Output } from "@angular/core";
+import { BsModalRef } from "ngx-bootstrap/modal";
+import { GlobalVarsService } from "../../../global-vars.service";
+import { BackendApiService, ProfileEntryResponse, TransactionFee } from "../../../backend-api.service";
+import { SwalHelper } from "../../../../lib/helpers/swal-helper";
+
+@Component({
+  selector: "admin-node-add-fees",
+  templateUrl: "./admin-node-add-fees.component.html",
+})
+export class AdminNodeAddFeesComponent implements OnInit {
+  @Input() txnType: string;
+  @Input() transactionFeeMap: { [k: string]: TransactionFee[] };
+
+  @Output() feeAdded = new EventEmitter();
+  publicKey: string;
+  feeAmount: number;
+  selectedCreator: ProfileEntryResponse;
+  updatingAllTxns: boolean = false;
+  updatingTransactionFees: boolean = false;
+
+  constructor(
+    private backendApi: BackendApiService,
+    public globalVars: GlobalVarsService,
+    public bsModalRef: BsModalRef
+  ) {}
+
+  ngOnInit() {
+    if (this.txnType === "ALL") {
+      this.updatingAllTxns = true;
+    }
+  }
+
+  setTransactionFees(): void {
+    if (!this.feeAmount || !this.selectedCreator?.PublicKeyBase58Check || this.updatingTransactionFees) {
+      return;
+    }
+    let newTransactionFees: TransactionFee[] = [];
+    // If we're only updating one transaction type, we use the existing list of transaction fees for this type.
+    // If we're updating the fee for all transactions, we overwrite.
+    if (!this.updatingAllTxns) {
+      newTransactionFees = this.transactionFeeMap[this.txnType] || [];
+    }
+
+    newTransactionFees = newTransactionFees.concat({
+      PublicKeyBase58Check: this.selectedCreator.PublicKeyBase58Check,
+      AmountNanos: this.feeAmount * 1e9,
+    });
+
+    SwalHelper.fire({
+      target: this.globalVars.getTargetComponentSelector(),
+      html: `${
+        (this.updatingAllTxns && " WARNING: this will overwrite all existing transaction fees you have set.\n") || ""
+      } Add fee of ${this.feeAmount} DESO for ${
+        this.selectedCreator?.Username || this.selectedCreator.PublicKeyBase58Check
+      }
+        for all${(!this.updatingAllTxns && " " + this.txnType) || ""} transactions.`,
```

Comment:
I think in general, we want to stay away from really complex in-line statements like this.  Especially interpolation like, "{(this.updatingAllTxns && " WARNING: this will overwrite all existing transaction fees you have set.\n") || "" }" that mixes logic with string creation.  I would prefer more verbose code that is easier to parse vs. more succinct code that is harder to parse.  @maebeam, @diamondhands0, I'd be curious to get your opinions on this so we can all be on the same page?

Reply:
yeah sounds good - I was messing around with what I wanted here. I don't intend for it to look this funky. There's be a clear if statement that just says `if updatingAllTxns -> stringA, else  stringB`

Reply:
Sounds good!